renovate.json

```diff
@@ -14,5 +14,12 @@
     "enabled": true,
     "ignoreTests": true,
     "platformAutomerge": true
+  },
+  "packageRules": [
```

Comment:
```
{
    "description": "Pin Python version in pyproject.toml to the latest 3.12.x patch",
    "matchPackageNames": ["python"],
    "matchManagers": ["poetry"],
    "allowedVersions": ">=3.12.0 <3.13.0"
}
```

This should pin your Python version in your Poetry config to what you are looking for.